### PR TITLE
chore(CPlusPlus): add prefix sum

### PR DESCRIPTION
**Pull request by Deeps-01**
**PR Checklist:**
- [ ] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [ ] My addition is on refer on the language README.md file
- [ ] My addition does not have a spelling problem
- [ ] My submission has a proper and user-friendly description of the algorithm
- [ ] My submission has the time complexity of the algorithm
- [ ] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

**Other information:**

### CODE REVIEW

File: algorithms/CPlusPlus/Arrays/prefix_sum_algorithm_implementation.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    
+    vector<long> arr;
+    
+    long max=0;
+    
+    for(int i=0; i<n ; i++ )
+    {
+      arr.push_back(0);
+     }
+
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Arrays/prefix_sum_algorithm_implementation.cpp:43]: fix indentation.

File: algorithms/CPlusPlus/Arrays/prefix_sum_algorithm_implementation.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    
+    for(int i=0; i<n ; i++ )
+    {
+       x = x+ arr[i] ;                        
+       
+       if(x>max)
+         max=x;
+     }
+       cout<<max;
+     }
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Arrays/prefix_sum_algorithm_implementation.cpp:63]: fix indentation.

File: algorithms/CPlusPlus/Arrays/prefix_sum_algorithm_implementation.cpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      
+    
+    
+
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Arrays/prefix_sum_algorithm_implementation.cpp:67]: Add examples/test-cases with time complexity

**Review by aayushjain7 — CHANGES_REQUESTED** [review]
Also, update the readme.MD file

**Comment by ming-tsai** [comment]
Hi @Deeps-01,
Any updates?

**Comment by ming-tsai** [comment]
Hi @Deeps-01,
This pull request will be closed when making a change please re-open this pull request.

**Pull request closed without merging**